core/iwasm/interpreter/wasm_interp_classic.c

```diff
@@ -1783,14 +1783,18 @@ wasm_interp_call_func_bytecode(WASMModuleInstance *module,
                 global = globals + global_idx;
                 global_addr = get_global_addr(global_data, global);
                 aux_stack_top = *(uint32 *)(frame_sp - 1);
-                if (aux_stack_top <= exec_env->aux_stack_boundary.boundary) {
-                    wasm_set_exception(module, "wasm auxiliary stack overflow");
-                    goto got_exception;
-                }
-                if (aux_stack_top > exec_env->aux_stack_bottom.bottom) {
-                    wasm_set_exception(module,
-                                       "wasm auxiliary stack underflow");
-                    goto got_exception;
+                if (wasm_exec_env_is_aux_stack_managed_by_runtime(exec_env)) {
```

Comment:
Do we need to apply similar changes for LLVM AOT/JIT and Fast JIT, refer to:

https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/compilation/aot_emit_variable.c#L192

https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/fast-jit/fe/jit_emit_variable.c#L264-L279

Reply:
Yes we do; however, for now we focus on the interpreter and we'll work on JIT/AOT next.

Reply:
OK, so let's merge this PR first?

Reply:
Yes, we can merge this one and next we'll continue working on AOT and JIT

Reply:
@wenyongh @yamt for AOT, should we disable the aux stack globally when WASI threads are enabled? by replacing this 
https://github.com/bytecodealliance/wasm-micro-runtime/blob/21ba730de38672c2eebf1d60c7e98ab2dbf87501/core/iwasm/compilation/aot_llvm.c#L1614

with something like
```
#if WASM_ENABLE_LIB_WASI_THREADS != 0
        comp_ctx->enable_aux_stack_check = false;
#else
        comp_ctx->enable_aux_stack_check = true;
#endif
```

Reply:
i guess it can be a workaround.
but i think it's better to avoid making the compiler depend on build-time configurations like this.

Reply:
at this point, should we just specify in the documentation to compile with `--disable-aux-stack-check` when using wasi threads?